doc/api/deprecations.md

```diff
@@ -0,0 +1,573 @@
+# Deprecated APIs
+
+Node.js may deprecate APIs when either: (a) use of the API is considered to be
+unsafe, (b) an improved alternative API has been made available, or (c)
+breaking changes to the API are expected in a future major release.
+
+Node.js utilizes two kinds of Deprecations:
+
+* Documentation-only
+* Runtime
```

Comment:
Looking at it, this could certainly be made clearer and I'll fix that.. but, the idea is that Runtime deprecation *also* includes a deprecation notice in the docs.